### ISSUE TITLE
Why set masking inputs to False for "response" tuning?

### ISSUE THREAD

**Issue by timothylimyl** [Lightning-AI/lit-llama#232]
When preparing the fine-tuning/instruction-tuning datasets such as in `prepare_alpaca/dolly.py`, why do we opt to not mask the inputs?  The comments mentioned: `mask_inputs: bool = False,  # as in alpaca-lora` but intuitively wouldn't masking the inputs make sense as we are just aiming to fine-tune the appropriate response given a particular instruction?

**Comment by awaelchli** [comment]
I agree conceptually it would make sense. I think that while reproducing the results and debugging, I just ended up going with exactly the same settings as in the other code repos and that worked in the end. For example here: https://github.com/tloen/alpaca-lora/blob/8bb8579e403dc78e37fe81ffbb253c413007323f/finetune.py#LL49C5-L49C20 they also have train_on_inputs=True (which corresponds to our mask_inputs=False). I didn't go back and check whether it makes a difference. But we could try!

**Comment by timothylimyl** [comment]
thanks for the reply, I will close this issue. I guess more discussion should take place either in discord or the newer lit-gpt repo.
